client/Assets/Scripts/Battle.cs

```diff
@@ -55,6 +62,33 @@ private void SetupInitialState()
         showInterpolationGhosts = false;
     }
 
+    private IEnumerator InitializeProjectiles()
+    {
+        yield return new WaitUntil(() => SocketConnectionManager.Instance.players.Count > 0);
+        SetInitialProjectile();
+    }
+
+    void SetInitialProjectile()
+    {
+        foreach (GameObject player in SocketConnectionManager.Instance.players)
+        {
+            // Now: we can access any skill from each player and choose the projectile this skill was assigned
+            // Problem: Instead of only using one skill to set all, we should be able to make a pool for each skill if they have different projectiles
+            // TODO: make it so that each skill has its own projectile
+            // for this, I need to keep on working in both this and ProjectileHandler scripts
+            skillProjectiles = player
+                .GetComponents<Skill>()
+                .Select(p => p.GetProjectileFromSkill())
```

Comment:
I believe this looks more clear.
```suggestion
                .Select(skill => skill.GetProjectileFromSkill())
```